concent_api/gatekeeper/views.py

```diff
@@ -31,14 +32,16 @@ def upload(request):
         return gatekeeper_access_denied_response('Unsupported content type.')
 
     path_to_file = request.get_full_path().partition(reverse('gatekeeper:upload'))[2]
-    response = parse_headers(request, path_to_file)
-    if response is not None:
-        logger.info(json.loads(response.content.decode())['message'])
-        return response
+    response_body = parse_headers(request, path_to_file)
```

Comment:
This name is confusing because the function does not necessarily return a response. If it does, it's a full response not just body. If it does not, it's a `FileInfo` object.

1) Call it `response_or_file_info` for clarity.
2) Add type annotation on `parse_headers()` that indicates what it returns (Please ask @rwrzesien for help if you're not familiar with the syntax).

Reply:
Still no annotation for the return type. You only added annotations for the parameters.